compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/bir/BIRGen.java

```diff
@@ -282,6 +288,38 @@ private void visitBuiltinFunctions(BLangPackage pkgNode, BLangFunction function)
         }
     }
 
+    private void visitMockFunctions(BIRPackage birPkg, Map<String, String> mockFunctionMap) {
+        for (BIRFunction function : birPkg.functions) {
+            List<BIRBasicBlock> functionBasicBlocks = function.basicBlocks;
+            for (BIRBasicBlock functionBasicBlock : functionBasicBlocks) {
+                BIRTerminator bbTerminator = functionBasicBlock.terminator;
+                if (bbTerminator.kind.equals(InstructionKind.CALL)) {
+                    mockFunctionMap.forEach((k, v) -> {
+                        String[] mockInfo = k.split(MOCK_ANNOTATION_DELIMITER);
+                        if (mockInfo.length != 2) {
+                            return;
+                        }
+                        if (((BIRTerminator.Call) bbTerminator).name.getValue().equals(mockInfo[1])) {
+                            ((BIRTerminator.Call) bbTerminator).name = getMockFunctionName(v, birPkg);
```

Comment:
Looks like there are 4-nested loops here. Please try to reduce.

Reply:
I have reduced the complexity in this case by removing the need to go thru the `mockFunctionMap`. There are only 2 loops now